### PR TITLE
Revision History UI

### LINKED ISSUE #1078: Add new UI for access to revision history in UI
Solving #306 is a prerequisite; see [discussion](https://github.com/Kitware/dive/issues/306#issuecomment-975619767).

### PR DESCRIPTION
**Pull request by subdavis**
## Summary of changes

* Adds revision history contextual sidebar, and a button to open it which so far I really dislike.  Feedback needed.
* Clone or download annotations when a past revision is loaded.

Please forgive the random book cover instead of real data.

# Menu open, Latest selected

Screenshot from 2022-02-28 08-22-10

# Menu open, previous rev selected 

Screenshot from 2022-02-28 08-22-23

# Download menu

Screenshot from 2022-02-28 08-22-37

fixes #1078

### CODE REVIEW

File: client/dive-common/store/context.ts [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
+
+function getComponents() {
+  const components: Record<string, VueConstructor<Vue>> = {};
+  Object.values(componentMap).forEach((v) => {
+    components[v.component.name] = v.component;
+  });
+  return components;
+}
+
+function toggle(active: string | null) {
```
Comment (BryonLewis) [inline client/dive-common/store/context.ts:40]: This use to have some Developer handy guarding based on the `ContextType` being a `keyof typeof ComponentMap` should we give this a more sane error if you use an invalid toggle?  The error isn't too bad currently but you would have to work your way back to see that you messed up the component name.  Feel free to resolve if you think this is clear enough already.
For Reference if you use an invalid component name:
image

File: client/dive-common/use/useRequest.ts [hunk 2]
```diff
@@ -36,3 +37,43 @@ export default function useRequest() {
...
+    totalCount: 0,
+    offset: 0,
+    limit: 20,
+  });
+  const allPages = shallowRef([] as T[]);
+
+  function reset() {
+    paginationParams.totalCount = 0;
+    paginationParams.offset = 0;
+    paginationParams.limit = 10;
```
Comment (BryonLewis) [inline client/dive-common/use/useRequest.ts:53]: why do you default to 20 and reset to 10?

**Review by BryonLewis — CHANGES_REQUESTED** [review]
This is really awesome and the ability to clone will be a nice stop-gap until true rollback is implemented.

File: client/platform/web-girder/views/RevisionHistory.vue [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    >
+      <v-list-item
+        v-for="revision in revisions"
+        :key="revision.revision"
+        :input-value="revision.revision === revisionId"
+        @click="checkout(revision.revision)"
+      >
+        <v-list-item-content>
+          <v-list-item-title>
+            {{ revision.revision }}: {{ revision.description }} by
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/platform/web-girder/views/RevisionHistory.vue:86]: I'm not sure what to do about that without committing much greater sins.

**Comment by subdavis** [comment]
Switch to three line layout

Screenshot from 2022-03-23 16-04-16

**Review by BryonLewis — APPROVED** [review]
Two super minor things that we may just want to create issues for in the future and don't need to be apart of this PR:

Don't know if this first one is really an issue?
Context View is saved in the state so if you were to back out to Data Browser with it open and load a new dataset it will start with the contextView open (either Confidence Pairs or Revision History).  I don't think this needs to be fixed.

Selecting a revision which is higher than your load limit (20) will cause it to refresh and reset the scroll page.  My fault about the `loadNext()` being activated on reload.  I have a dataset with 40 revision, I click on revision 4 and it resets all of the data and scrolling so that it isn't highlighted in the list when the revision reloads.

**Pull request merged**